src/olympia/addons/tasks.py

```diff
@@ -31,31 +31,31 @@
 @task
 @use_primary_db
 def version_changed(addon_id, **kw):
-    update_last_updated(addon_id)
-    update_appsupport([addon_id])
-
-
-def update_last_updated(addon_id):
-    queries = Addon._last_updated_queries()
     try:
         addon = Addon.objects.get(pk=addon_id)
     except Addon.DoesNotExist:
         log.info(
             '[1@None] Updating last updated for %s failed, no addon found' % addon_id
         )
         return
+    log.info('[1@None] Updating last updated for %s.' % addon.pk)
+    addon.update(last_updated=compute_last_updated(addon))
+    update_appsupport([addon.pk])
 
-    log.info('[1@None] Updating last updated for %s.' % addon_id)
 
+def compute_last_updated(addon):
+    queries = Addon._last_updated_queries()
     if addon.status == amo.STATUS_APPROVED:
         q = 'public'
     else:
         q = 'exp'
```

Comment:
I'm aware that you're not touching this functionality but these `_last_updated_queries` are weird.  If the addon status isn't `STATUS_APPROVED` (e.g. `STATUS_NOMINATED`) then you get the "exp" query ("experimental"??) which excludes `STATUS_NOMINATED`.

Reply:
Yeah. I filed https://github.com/mozilla/addons-server/issues/16295